contrib/babelfishpg_tsql/src/catalog.c

```diff
@@ -2308,3 +2312,278 @@ get_db_owner_role_name(const char *dbname)
 	table_close(bbf_authid_user_ext_rel, RowExclusiveLock);
 	return db_owner_role;
 }
+
+void
+rename_update_bbf_catalog(RenameStmt *stmt)
+{
+	switch (stmt->renameType)
+	{	
+		case OBJECT_TABLE:
+			break;
+		case OBJECT_VIEW:
+			rename_view_update_bbf_catalog(stmt);
+			break;
+		case OBJECT_PROCEDURE:
+			rename_proc_update_bbf_catalog(stmt);
+			break;
+		case OBJECT_FUNCTION:
+			rename_func_update_bbf_catalog(stmt);
+			break;
+		default:
+			break;	
+	}
+}
+
+static void
+rename_view_update_bbf_catalog(RenameStmt *stmt)
+{
+	// update the 'object_name' in 'babelfish_view_def'
+	Relation		bbf_view_def_rel;
+	TupleDesc		bbf_view_def_dsc;
+	ScanKeyData		key[2];
+	HeapTuple		usertuple;
+	HeapTuple		new_tuple;
+	TableScanDesc		tblscan;
+	Datum			new_record_view_def[BBF_VIEW_DEF_NUM_COLS];
+	bool			new_record_nulls_view_def[BBF_VIEW_DEF_NUM_COLS];
+	bool			new_record_repl_view_def[BBF_VIEW_DEF_NUM_COLS];
+	char			*cur_db = get_cur_db_name();
+	char			*curr_schemaname = stmt->relation->schemaname;
+
+	// build the tuple to insert
+	MemSet(new_record_view_def, 0, sizeof(new_record_view_def));
+	MemSet(new_record_nulls_view_def, false, sizeof(new_record_nulls_view_def));
+	MemSet(new_record_repl_view_def, false, sizeof(new_record_repl_view_def));
+
+	// open the catalog table
+	bbf_view_def_rel = table_open(get_bbf_view_def_oid(), RowExclusiveLock);
+
+	// get the description of the table
+	bbf_view_def_dsc = RelationGetDescr(bbf_view_def_rel);
+
+	// serach for the row for update => build the key
+	ScanKeyInit(&key[0],
```

Comment:
I feel this is not correct. 
sys.babelfish_view_def's primary key is dbid, schema_name, object_name, 
plz refer to this search_bbf_view_def function